src/Microsoft.OData.Client/DataServiceContext.cs

```diff
@@ -2573,6 +2628,70 @@ public virtual void AddRelatedObject(object source, string sourceProperty, objec
             this.entityTracker.IncrementChange(targetResource);
         }
 
+        /// <summary>Adds a related object to the context and creates the link that defines the relationship between the two objects in a single request.</summary>
+        /// <param name="source">The parent object that is being tracked by the context.</param>
+        /// <param name="sourceProperty">The name of the navigation property that returns the related object based on an association between the two entities.</param>
+        /// <param name="target">The related object that is being added.</param>
+        /// /// <remarks>
+        /// The sourceProperty MUST be a single-value object, not a collection.
+        /// </remarks>
+        public virtual void SetRelatedObject(object source, string sourceProperty, object target)

```

Comment:
So the only difference between this and `AddRelatedObject` is that this method only supports single-value navigation properties whereas `AddRelatedObject` only supports collection navigation properties? Could these not have been unified into a single method or would that be a breaking change?

Though I don't think it's a deal-breaker, I think there's a potential to abstract the common parts into a common method since the methods are almost identical, all except 2 lines or so.

That said, the summaries of the methods don't make it more confusing since they are identical. I think we could move the distinction to the summary instead of being in the remarks section of the xml doc, since that restriction is the reason we have two separate methods instead of one.

I think it would be helpful to have names that make the distinction between the two clearer. Is it possible to consider a name that reflects that this method only applies to single value properties? Like `AddSingleRelatedObject`?

Reply:
On second thought, with the suggested `AddSingleRelatedObject`, one could also think that we are adding a single related object to a collection property. So it may not be sufficient to disambiguate between the two methods.

Reply:
I have updated the summary

Reply:
I think this is the one I was  asking. Cam this be used with   normal `SaveChanges` API call? Are there any implications as we don't post to single-valued navigation properties on a normal `POST`?